App/UI Crash when hitting the back button

**Describe the bug**
A crash occurs when clicking the back button. Possibly because there was no more history to return to.

**To Reproduce**
Steps to reproduce the behavior:

1. Create directory which begins with . (hidden directory)
2. Go up 1 directory and then into another directory (should be a sibling of the hidden directory created).
3. Select multiple items and use the "cut" to move them.
4. Click on the path showed for current directory.
5. Click the parent directory.
6. Go into previously created hidden directory and paste the files.
7. Click the android menu back button 3 times.

**Expected behavior**
Application would show a message stating "click back again to close application"

**Screenshots**
n/a

**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S22
 - OS: Android 13
 - Rooted: No
 - Version: 3.8.5

**Additional context**
I clicked the back button multiple times thinking i would end up in the folder from which i had cut the files from.

## Exception
* _App Name:_ Amaze File Manager
* _Package:_ com.amaze.filemanager
* _Version:_ 3.8.5
* _User Action:_ UI Error
* _Request:_ Application crash
* _OS:_ Linux Android 13 - 33
* _Device:_ r0q
* _Model:_ SM-S901U
* _Product:_ r0qsqw
<details><summary><b>Crash log </b></summary><p>

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
	at android.os.AsyncTask$4.done(AsyncTask.java:415)
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:250)
	at java.util.concurrent.FutureTask.run(FutureTask.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.amaze.filemanager.ui.fragments.data.MainFragmentViewModel.getFileCount()' on a null object reference
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.postListCustomPathProcess(LoadFilesListTask.java:271)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:193)
	at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.doInBackground(LoadFilesListTask.java:85)
	at android.os.AsyncTask$3.call(AsyncTask.java:394)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	... 3 more
```

</details>
<hr>

--- a/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
+++ b/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
@@ -286,7 +286,6 @@ public class LoadFilesListTask
     }
 
     Collections.sort(list, new FileListSorter(viewModel.getDsort(), sortBy, isAscending));
-
   }
 
   private @Nullable LayoutElementParcelable createListParcelables(HybridFileParcelable baseFile) {

--- a/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
+++ b/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
@@ -264,6 +264,11 @@ public class LoadFilesListTask
 
     MainFragmentViewModel viewModel = mainFragment.getMainFragmentViewModel();
 
+    if (viewModel == null) {
+      LOG.error("MainFragmentViewModel is null, this is a bug");
+      return;
+    }
+
     for (int i = 0; i < list.size(); i++) {
       LayoutElementParcelable layoutElementParcelable = list.get(i);
 
@@ -274,17 +279,14 @@ public class LoadFilesListTask
       }
 
       if (layoutElementParcelable.isDirectory) {
-        viewModel.setFolderCount(mainFragment.getMainFragmentViewModel().getFolderCount() + 1);
+        viewModel.incrementFolderCount();
       } else {
-        viewModel.setFileCount(mainFragment.getMainFragmentViewModel().getFileCount() + 1);
+        viewModel.incrementFileCount();
       }
     }
 
-    if (viewModel != null) {
-      Collections.sort(list, new FileListSorter(viewModel.getDsort(), sortBy, isAscending));
-    } else {
-      LOG.error("MainFragmentViewModel is null, this is a bug");
-    }
+    Collections.sort(list, new FileListSorter(viewModel.getDsort(), sortBy, isAscending));
+
   }
 
   private @Nullable LayoutElementParcelable createListParcelables(HybridFileParcelable baseFile) {

--- a/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
+++ b/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
@@ -251,15 +251,15 @@ public class LoadFilesListTask
 
   private void postListCustomPathProcess(
       @NonNull List<LayoutElementParcelable> list, @NonNull MainFragment mainFragment) {
-    int t = SortHandler.getSortType(context.get(), path);
-    int sortby;
-    int asc;
-    if (t <= 3) {
-      sortby = t;
-      asc = 1;
+
+    int sortType = SortHandler.getSortType(context.get(), path), sortBy, isAscending;
+
+    if (sortType <= 3) {
+      sortBy = sortType;
+      isAscending = 1;
     } else {
-      asc = -1;
-      sortby = t - 4;
+      isAscending = -1;
+      sortBy = sortType - 4;
     }
 
     MainFragmentViewModel viewModel = mainFragment.getMainFragmentViewModel();
@@ -268,6 +268,7 @@ public class LoadFilesListTask
       LayoutElementParcelable layoutElementParcelable = list.get(i);
 
       if (layoutElementParcelable == null) {
+        //noinspection SuspiciousListRemoveInLoop
         list.remove(i);
         continue;
       }
@@ -280,7 +281,7 @@ public class LoadFilesListTask
     }
 
     if (viewModel != null) {
-      Collections.sort(list, new FileListSorter(viewModel.getDsort(), sortby, asc));
+      Collections.sort(list, new FileListSorter(viewModel.getDsort(), sortBy, isAscending));
     } else {
       LOG.error("MainFragmentViewModel is null, this is a bug");
     }

--- a/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
+++ b/app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/LoadFilesListTask.java
@@ -252,7 +252,9 @@ public class LoadFilesListTask
   private void postListCustomPathProcess(
       @NonNull List<LayoutElementParcelable> list, @NonNull MainFragment mainFragment) {
 
-    int sortType = SortHandler.getSortType(context.get(), path), sortBy, isAscending;
+    int sortType = SortHandler.getSortType(context.get(), path);
+    int sortBy;
+    int isAscending;
 
     if (sortType <= 3) {
       sortBy = sortType;

--- a/app/src/main/java/com/amaze/filemanager/ui/fragments/data/MainFragmentViewModel.kt
+++ b/app/src/main/java/com/amaze/filemanager/ui/fragments/data/MainFragmentViewModel.kt
@@ -243,4 +243,12 @@ class MainFragmentViewModel : ViewModel() {
         }
         return mutableLiveData
     }
+
+    fun incrementFileCount() {
+        fileCount++
+    }
+
+    fun incrementFolderCount() {
+        folderCount++
+    }
 }

--- a/app/src/main/java/com/amaze/filemanager/ui/fragments/data/MainFragmentViewModel.kt
+++ b/app/src/main/java/com/amaze/filemanager/ui/fragments/data/MainFragmentViewModel.kt
@@ -244,10 +244,16 @@ class MainFragmentViewModel : ViewModel() {
         return mutableLiveData
     }
 
+    /**
+     * increments `fileCount`
+     */
     fun incrementFileCount() {
         fileCount++
     }
 
+    /**
+     * increments `folderCount`
+     */
     fun incrementFolderCount() {
         folderCount++
     }

